### ISSUE TITLE
FreePBX GUI can not connect to DB

### ISSUE THREAD

**Issue by fukemy** [ugoviti/izpbx#65]
Hi, after commend the line: `network_mode: host` inside `docker_composer`, I success to open FreePBX GUI after deploy izpbx, but I got this error, can you help?

<img width="1512" alt="Ảnh màn hình 2024-04-09 lúc 11 41 17" src="https://github.com/ugoviti/izpbx/assets/8202437/caac6e05-8626-4e2b-818e-4aca5c4c803b">

```

UNIQUE_ID | "ZhTG4dIBm6lz4VDqeZTYBAAAAAE"
-- | --
HTACCESS | "on"
HTTP_HOST | "localhost"
HTTP_CONNECTION | "keep-alive"
HTTP_CACHE_CONTROL | "max-age=0"
HTTP_SEC_CH_UA | ""Chromium";v="118", "Google Chrome";v="118", "Not=A?Brand";v="99""
HTTP_SEC_CH_UA_MOBILE | "?0"
HTTP_SEC_CH_UA_PLATFORM | ""macOS""
HTTP_UPGRADE_INSECURE_REQUESTS | "1"
HTTP_USER_AGENT | "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36"
HTTP_ACCEPT | "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"
HTTP_SEC_FETCH_SITE | "none"
HTTP_SEC_FETCH_MODE | "navigate"
HTTP_SEC_FETCH_USER | "?1"
HTTP_SEC_FETCH_DEST | "document"
HTTP_ACCEPT_ENCODING | "gzip, deflate, br"
HTTP_ACCEPT_LANGUAGE | "vi-VN,vi;q=0.9,en-US;q=0.8,en;q=0.7,fr-FR;q=0.6,fr;q=0.5"
HTTP_COOKIE | "mailpoet_subscriber=%7B%22subscriber_id%22%3A1%7D; mailpoet_page_view=%7B%22timestamp%22%3A1710488431%7D; PHPSESSID=ge5ichlgijocli53jn5rtoesv7"
PATH | "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
SERVER_SIGNATURE | ""
SERVER_SOFTWARE | "Apache/2.4.37 (rocky) OpenSSL/1.1.1k PHP/7.4.33"
SERVER_NAME | "localhost"
SERVER_ADDR | "192.168.16.3"
SERVER_PORT | "80"
REMOTE_ADDR | "192.168.65.1"
DOCUMENT_ROOT | "/var/www/html"
REQUEST_SCHEME | "http"
CONTEXT_PREFIX | ""
CONTEXT_DOCUMENT_ROOT | "/var/www/html"
SERVER_ADMIN | "root@localhost"
SCRIPT_FILENAME | "/var/www/html/admin/config.php"
REMOTE_PORT | "22622"
GATEWAY_INTERFACE | "CGI/1.1"
SERVER_PROTOCOL | "HTTP/1.1"
REQUEST_METHOD | "GET"
QUERY_STRING | ""
REQUEST_URI | "/admin/config.php"
SCRIPT_NAME | "/admin/config.php"
PHP_SELF | "/admin/config.php"
REQUEST_TIME_FLOAT | 1712637665.4823
REQUEST_TIME | 1712637665

```

**Comment by ugoviti** [comment]
If you disable network mode host the PBX doesn't works as expeceted (RTP ports to open)

do you see the comments inside .env file?

```
## WARNING: if docker-compose is configured with "network_mode: host" then use "MYSQL_SERVER=127.0.0.1" or the address of the remote database server
## WARNING: if docker-compose is not configured with "network_mode: host" then use "MYSQL_SERVER=db" or the address of the remote database server
#MYSQL_SERVER=db
MYSQL_SERVER=127.0.0.1
MYSQL_DATABASE=asterisk
MYSQL_DATABASE_CDR=asteriskcdrdb
MYSQL_USER=asterisk
```

I suggest you to clean all, and restart from scratch without changing nothing in the .env for the first run.

When you seed that can connect to the FreePBX GUI, you can customize your setup (change password, etc.)

Kind Regards

**Comment by ugoviti** [comment]
Follow the README.md please:

# Deploy izPBX
Using **docker compose** is the suggested method:

- Install your prefered Linux OS into VM or Baremetal Server

- Install Docker Runtime with `docker compose` plugin for your Operating System from https://www.docker.com/get-started
  - RHEL8 based distro Quick&Dirty commands (skip if you use other distribution):
```
sudo dnf config-manager --add-repo=https://download.docker.com/linux/centos/docker-ce.repo
sudo dnf install docker-ce docker-compose-plugin -y
sudo systemctl enable --now docker
```

- Create a `docker-compose.yml`, or clone git repository, or download latest tarbal release from: https://github.com/ugoviti/izpbx/releases and unpack it into a directory (ex. `/opt/izpbx`), faster method with git:
  - `git clone https://github.com/ugoviti/izpbx.git /opt/izpbx`
  - `cd /opt/izpbx`

- Checkout into latest official release:
  - `git checkout tags/$(git tag | sort --version-sort | tail -1)`

- Copy default configuration file `default.env` into `.env`:
  - `cp default.env .env`

- Customize `.env` variables, specially the security section of default passwords:
  - `vim .env`

- Deploy and start izpbx using `docker compose` command:
  - `docker compose up -d`

- Wait the pull to finish (~60 seconds with fast internet connections) and point your web browser to the IP address of your docker host and follow initial setup guide

**Note:** by default, to correctly handle SIP NAT and SIP-RTP UDP traffic, the izpbx container will use the `network_mode: host`, so the izpbx container will be exposed directly to the outside network without using docker internal network range (**network_mode: host** will prevent multiple izpbx containers from running inside the same host).  
Modify docker-compose.yml and comment `#network_mode: host` if you want run multiple izpbx containers in the same host (not production tested. There will be problems with RTP traffic).
Another available option is to disable `network_mode: host` and use **macvlan** network mode used for running izPBX into multi-tenant mode.

**Comment by ugoviti** [comment]
Thankyou, fixed upstream the version to 20.16.12

**Comment by meng7171** [comment]
TQ for the INFO of 'use macvlan network mode '

https://github.com/ugoviti/izpbx/issues/65#issuecomment-2051464169
